src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+    D-Bus API.
+    """
+
+    def __init__(self, config):
+        """Initializes the module by choosing devices and assembling a table of connection
+        IDs.
+
+        config: The configuration dictionary built in netcheck-daemon.
+        """
+
+        self.logger = logging.getLogger(__name__)
+        self.network_activation_timeout = config['network_activation_timeout']
+        self.wired_network_name = config['wired_network_name']
+        self.wifi_network_names = config['wifi_network_names']
+
+        self.connection_id_to_connection_dict = self._build_connection_id_table()
+
+        self.wired_device, self.wireless_device = self._create_device_objects(
+            config['wired_interface_name'],
+            config['wifi_interface_name'])
+
+    def activate_connection(self, connection_id):
+        """Tells NetworkManager to activate a connection with the supplied connection_id.
+        Returns True if there are no errors, False otherwise.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        success = False
+
+        if not self.connection_is_activated(connection_id):
+            connection = self.connection_id_to_connection_dict[connection_id]
+            network_device = self._get_device_for_connection(connection)
+
+            networkmanager_output = NetworkManager.NetworkManager.ActivateConnection(
+                connection, network_device, '/')
+
+            try:
+                self._run_proxy_call(networkmanager_output)
+                success = self._wait_for_connection(connection)
+
+            except Exception as exception:
+                self.logger.error('D-Bus call failed: %s', exception)
+
+        else:
+            self.logger.debug('Connection %s is already activated.', connection_id)
+            success = True
+
+        return success
+
+    def get_connection_ip(self, connection_id):
+        """Attempts to retrieve the IP address associated with the given connection's gateway.
+        If the IP address is unable to be retrieved, None is returned.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        # I decided not to throw an exception here because the intended caller would end up
+        #   simply using it for flow control.
+        ip_address = None
+
+        connection = self.connection_id_to_connection_dict[connection_id]
+
+        if self.connection_is_activated(connection_id):
+            device = self._get_device_for_connection(connection)
+            gateway_octets = device.Ip4Config.Gateway.split('.')
+
+            for address_data in device.Ip4Config.AddressData:
+                address_octets = address_data['address'].split('.')
+                if address_octets[0] == gateway_octets[0] \
+                        and address_octets[1] == gateway_octets[1] \
+                        and address_octets[2] == gateway_octets[2]:
```

Comment:
(This might be the only real error.) Don't assume the subnet has 256 addresses. Use the subnet mask to determine the size of the subnet.